gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -507,9 +513,10 @@ public void apply(JobListener jobListener, JobContext jobContext)
         }
 
         // Perform work needed before writing is done
-        Boolean canCleanUp = this.canCleanStagingData(this.jobContext.getJobState());
-        workUnitStream = closer.register(new DestinationDatasetHandlerService(jobState, canCleanUp, this.eventSubmitter))
-            .executeHandlers(workUnitStream);
+        this.canCleanUpStagingData = this.canCleanStagingData(this.jobContext.getJobState());
```

Comment:
Is this from other change? seems not related to this PR

Reply:
this is actually needed and may easily get overlooked. It's a process that make sure the workUnit can handle shards for target directory. Took me a while the figure out and debug when testing in the cluster...